examples/all-clusters-app/all-clusters-common/src/WhmManufacturer.cpp

```diff
@@ -0,0 +1,270 @@
+
+using namespace chip;
+
+using Protocols::InteractionModel::Status;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace WaterHeaterManagement {
+
+CHIP_ERROR WhmManufacturer::Init()
+{
+    WaterHeaterManagementDelegate * dg = GetWhmManufacturer()->GetWhmDelegate();
+    if (dg == nullptr)
+    {
+        ChipLogError(AppServer, "WhmDelegate is not initialized");
+        return CHIP_ERROR_UNINITIALIZED;
+    }
+
+    mBoostActive = false;
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR WhmManufacturer::Shutdown()
+{
+    return CHIP_NO_ERROR;
+}
+
+BitMask<WaterHeaterDemandBitmap> WhmManufacturer::DetermineHeatingSources()
+{
+    WaterHeaterManagementDelegate * dg = GetWhmManufacturer()->GetWhmDelegate();
+    if (dg == nullptr)
+    {
+        ChipLogError(AppServer, "WhmDelegate is not initialized");
+        return BitMask<WaterHeaterDemandBitmap>(0);
+    }
+
+    // A list of valid heaterTypes
+    uint8_t waterHeaterTypeValues[] = {
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kImmersionElement1),
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kImmersionElement2),
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kHeatPump),
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kBoiler),
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kOther),
+    };
+
+    // The corresponding list of valid headerDemands
+    uint8_t waterHeaterDemandValues[] = {
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kImmersionElement1),
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kImmersionElement2),
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kHeatPump),
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kBoiler),
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kOther),
+    };
+
+    // Iterate across the valid waterHeaterTypes seeing which heating sources are available based on heaterTypes.
+    // Set the corresponding bit in the heaterDemand bitmap.
+    BitMask<WaterHeaterTypeBitmap> heaterTypes = dg->GetHeaterTypes();
+
+    uint8_t heaterDemandMask = 0;
+    for (uint8_t idx = 0; idx < sizeof(waterHeaterTypeValues) / sizeof(waterHeaterTypeValues[0]); idx++)
+    {
+        // Is this heating source being used?
+        if (heaterTypes.Raw() & waterHeaterTypeValues[idx])
+        {
+            heaterDemandMask |= waterHeaterDemandValues[idx];
+        }
+    }
+
+    return BitMask<WaterHeaterDemandBitmap>(heaterDemandMask);
+}
+
+Status WhmManufacturer::TurnHeatingOn()
+{
+    Status status = Status::Success;
+
+    ChipLogProgress(AppServer, "WhmManufacturer::TurnHeatingOn");
+
+    WaterHeaterManagementDelegate * dg = GetWhmDelegate();
+
+    if (dg->GetBoostState() == BoostStateEnum::kActive)
+    {
+        mBoostActive = true;
+    }
+
+    return status;
+}
+
+Status WhmManufacturer::TurnHeatingOff()
+{
+    Status status = Status::Success;
+
+    ChipLogProgress(AppServer, "WhmManufacturer::TurnHeatingOff");
+
+    if (mBoostActive)
+    {
+        mBoostActive = false;
```

Comment:
Again shouldn't this be a callback to actually turn something on/off (rather than setting mBoostActive = False.

Reply:
See above